src/imitation/rewards/discrim_net.py

```diff
@@ -75,6 +63,79 @@ def build_graph(self):
     FIXME(sam): remove train reward/test reward distinction.
     """
 
+  def _add_default_train_stats(self) -> None:
+    """Updates `self._train_stats` with several default summary Tensors.
+
+    Also creates a `tf.summary.histogram` for the `disc_logits`
+    argument.
+
+    Params:
+      disc_logits: A rank-1 Tensor containing binary logit for expert
+        (positive value) versus generator (negative values).
+    """
+    tf.summary.histogram("disc_logits", self._disc_logits)
+    bin_is_generated_pred = self._disc_logits > 0
+    bin_is_generated_true = self._labels_ph > 0
+    bin_is_expert_true = tf.logical_not(bin_is_generated_true)
+    int_is_generated_pred = tf.cast(bin_is_generated_pred, tf.int32)
+    int_is_generated_true = tf.cast(bin_is_generated_true, tf.int32)
+    n_generated = tf.reduce_sum(int_is_generated_true)
+    n_labels = tf.size(self._labels_ph)
+    n_expert = n_labels - n_generated
+    pct_expert = tf.cast(n_expert, tf.float32) / tf.cast(n_labels, tf.float32)
+    n_expert_pred = (tf.size(bin_is_generated_pred)
+                     - tf.reduce_sum(int_is_generated_pred))
+    pct_expert_pred = (tf.cast(n_expert_pred, tf.float32)
+                       / tf.cast(n_labels, tf.float32))
+    correct_vec = tf.equal(bin_is_generated_pred, bin_is_generated_true)
+    acc = tf.reduce_mean(tf.cast(correct_vec, tf.float32))
+
+    _n_pred_expert = tf.reduce_sum(tf.cast(
+      tf.logical_and(bin_is_expert_true, correct_vec), tf.float32))
+    _n_expert_or_1 = tf.cast(tf.maximum(1, n_expert), tf.float32)
+    expert_acc = _n_pred_expert / _n_expert_or_1
+
+    _n_pred_gen = tf.reduce_sum(tf.cast(
+      tf.logical_and(bin_is_generated_true, correct_vec), tf.float32))
+    _n_gen_or_1 = tf.cast(tf.maximum(1, n_generated), tf.float32)
+    generated_acc = _n_pred_gen / _n_gen_or_1
+
+    label_dist = tf.distributions.Bernoulli(logits=self._disc_logits)
+    entropy = tf.reduce_mean(label_dist.entropy())
+
+    self._train_stats.update([
+      # basic xent loss
+      ('disc_xent', tf.reduce_mean(self._disc_loss)),
```

Comment:
Is using `_disc_loss` always  correct? What if there's a regulariser or something in the loss? Might be safer just to use `label_dist.cross_entropy(self._disc_logits)` in this part (but also make sure the full discriminator loss is logged somewhere, maybe under a name like `disc_loss` instead).

Reply:
Ok, I am explicitly constructing the xent now.